src/modinfo.c

```diff
@@ -2624,6 +2626,14 @@ sr_modinfo_data_store(struct sr_mod_info_s *mod_info)
                 }
 
                 if (mod_info->ds == SR_DS_RUNNING) {
+                    /* Acquire write lock on oper ds */
+                    shm_lock = &mod->shm_mod->data_lock_info[SR_DS_OPERATIONAL];
+
+                    if ((err_info = sr_rwlock(&shm_lock->data_lock, SR_MOD_LOCK_TIMEOUT,
+                            SR_LOCK_WRITE, mod_info->conn->cid, __func__, sr_shmmod_recover_cb, NULL))) {
```

Comment:
I am not entirely sure about this invocation here.. Is it okay to call this with `NULL` for recovery?

Reply:
Just look at the code of `sr_shmmod_recover_cb`, obviously not. I have not noticed it.